pipeline.yaml

```diff
@@ -0,0 +1,85 @@
+resources:
+  - name: source
+    type: git
+    source:
+      uri: ((git_source))
+      private_key: ((privatekey))
+      branch: ((git_branch))
+      skip_ssl_verification: ((git_skip_ssl_verification))
+      # This enables triggering on tag changes that match the glob pattern
+      # tag_filter: "*"
+
+  - name: bb-gitversion
+    type: docker-image
+    source:
+      repository: elbb/bb-gitversion
+      tag: ((bb-gitversion-version))
+
+  - name: s3-gitversion
+    type: s3
+    source:
+      bucket: ((name))
+      endpoint: ((s3_endpoint))
+      # at least one regex capture group is needed here
+      regexp: version/(gitversion).tar.gz
+      access_key_id: ((s3_access_key))
+      secret_access_key: ((s3_secret_key))
+      disable_ssl: ((s3_disable_ssl))
+
+  - name: aws-cli
+    type: docker-image
+    source:
+      repository: amazon/aws-cli
+      tag: latest
+
+jobs:
+  - name: Generate gitversion and put it on S3
+    public: true
+    plan:
+      - in_parallel:
+          - get: bb-gitversion
+          - get: source
+          - get: aws-cli
+      - task: Generate version
+        image: bb-gitversion
+        config:
+          platform: linux
+          inputs:
+            - name: source
+          outputs:
+            - name: gen
+          run:
+            path: /bin/bash
+            args:
+              - -exc
+              - |
+                entrypoint.sh && 
+                cat gen/json/gitversion.json && 
+                tar cvfz gen/gitversion.tar.gz gen/*
+          params:
+            GIT_PATH: source
+            GEN_PATH: gen
+      - task: Check/Create bucket for ((name)) on S3
+        image: aws-cli
+        config:
+          platform: linux
+          run:
+            path: /bin/sh
+            args:
+              - -xc
+              - |
+                mkdir -p ~/.aws
+                # hide setting credentials in plain text in concourse ui
+                set +x
+                export TERM="xterm-256color"
+                echo "[default]" > ~/.aws/credentials
+                echo "aws_access_key_id=((s3_access_key))" >> ~/.aws/credentials
+                echo "aws_secret_access_key=((s3_secret_key))" >> ~/.aws/credentials
+                set -x
+                # check if bucket exists on s3 endpoint. If not create one
+                aws --endpoint-url ((s3_endpoint)) s3api head-bucket --bucket ((name)) 2> /dev/null
+                if [ $? != 0 ]; then aws s3 mb s3://((name)) --endpoint-url ((s3_endpoint)); fi
+      - put: s3-gitversion
+        params:
+          file: gen/gitversion.tar.gz
```

Comment:
we overwrite this file on each pipeline run ... this is okay for manual triggering when there is no version change.
But what on version change ... ? Either the version information itself should be part of the bucketname or the version file.

Reply:
we aggreed to use the version as part of the bucketname

Reply:
S3 bucket names will now be created with SemVer suffix e.g. bb-example-1.2.3

Reply:
After further consideration the s3 bucket will be created without the version suffix. Instead, the generated gitversion archive file gets either the FullSemVer suffix for `master` and a InformationalVersion as suffix. These Version strings are slightly modified to allow the usage as docker tags ('+' replaced by '-') .